streams/src/main/resources/common/message/SubscriptionInfo.json

```diff
@@ -0,0 +1,71 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+{
+  "name": "SubscriptionInfo",
+  "validVersions": "1-5",
+  "fields": [
+    {
+      "name": "version",
```

Comment:
Why do we need a separate version field?

Reply:
AFAICT, we need it for two reasons:
1. compatibility with the hand-coded serialized format, which would have to inter-operate with the generated schemas during a rolling bounce
2. When we use the generated schemas, we have to know a priori what version we're using. If we don't serialize it in the message itself, how can we know?

Note that if problem 1 is the only concern, then we can deprecate this field starting at version 6 and save ourselves 32 bits. So problem 2 is the important question to answer. I couldn't figure out a way around this... Can you?